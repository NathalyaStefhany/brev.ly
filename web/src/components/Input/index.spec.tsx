import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Input } from '.';

describe('Input tests', () => {
  it('should render correctly', () => {
    expect(
      render(<Input id="input-test" label="Title" placeholder="Placeholder" />),
    );
  });

  it('should display error when provided', () => {
    render(
      <Input
        id="input-test"
        label="Title"
        placeholder="Placeholder"
        error="Error message"
      />,
    );

    expect(screen.getByTestId('input-test-error')).toBeInTheDocument();
    expect(screen.getByTestId('input-test-error')).toHaveTextContent(
      'Error message',
    );
  });

  it('should not display error', () => {
    render(<Input id="input-test" label="Title" placeholder="Placeholder" />);

    expect(screen.queryByTestId('input-test-error')).not.toBeInTheDocument();
  });

  it('should display input correctly', () => {
    const { container } = render(
      <Input id="input-test" label="Title" placeholder="Placeholder" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should display input with error correctly', () => {
    const { container } = render(
      <Input
        id="input-test"
        label="Title"
        placeholder="Placeholder"
        error="Error message"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
