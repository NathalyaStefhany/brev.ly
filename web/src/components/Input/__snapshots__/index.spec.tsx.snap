// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input tests > should display input correctly 1`] = `
<div>
  <div
    class="relative max-w-176 w-full flex flex-col-reverse gap-4"
  >
    <input
      aria-describedby="input-test-error"
      aria-invalid="false"
      class="peer z-1 text-md text-gray-600 font-normal border border-gray-300 caret-blue-base rounded-lg py-7 placeholder:text-gray-400 focus:outline-blue-base"
      id="input-test"
      placeholder="Placeholder"
      style="padding-left: calc(1rem); padding-right: 1rem;"
      type="text"
    />
    <label
      class="text-xs text-gray-500 uppercase focus:font-bold peer-focus:font-bold peer-focus:text-blue-base"
      for="input-test"
    >
      Title
    </label>
  </div>
</div>
`;

exports[`Input tests > should display input with error correctly 1`] = `
<div>
  <div
    class="relative max-w-176 w-full flex flex-col-reverse gap-4"
  >
    <div
      class="flex flex-row items-center gap-4"
    >
      <svg
        fill="var(--color-danger)"
        height="1rem"
        viewBox="0 0 256 256"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M236.8,188.09,149.35,36.22h0a24.76,24.76,0,0,0-42.7,0L19.2,188.09a23.51,23.51,0,0,0,0,23.72A24.35,24.35,0,0,0,40.55,224h174.9a24.35,24.35,0,0,0,21.33-12.19A23.51,23.51,0,0,0,236.8,188.09ZM222.93,203.8a8.5,8.5,0,0,1-7.48,4.2H40.55a8.5,8.5,0,0,1-7.48-4.2,7.59,7.59,0,0,1,0-7.72L120.52,44.21a8.75,8.75,0,0,1,15,0l87.45,151.87A7.59,7.59,0,0,1,222.93,203.8ZM120,144V104a8,8,0,0,1,16,0v40a8,8,0,0,1-16,0Zm20,36a12,12,0,1,1-12-12A12,12,0,0,1,140,180Z"
        />
      </svg>
      <span
        class="text-sm text-gray-500"
        data-testid="input-test-error"
        id="input-test-error"
      >
        Error message
      </span>
    </div>
    <input
      aria-describedby="input-test-error"
      aria-invalid="true"
      class="peer z-1 text-md text-gray-600 font-normal border border-gray-300 caret-blue-base rounded-lg py-7 placeholder:text-gray-400 focus:outline-danger"
      id="input-test"
      placeholder="Placeholder"
      style="padding-left: calc(1rem); padding-right: 1rem;"
      type="text"
    />
    <label
      class="text-xs text-gray-500 uppercase focus:font-bold peer-focus:font-bold peer-focus:text-danger"
      for="input-test"
    >
      Title
    </label>
  </div>
</div>
`;

exports[`Input tests > should display input with fixed placeholder correctly 1`] = `
<div>
  <div
    class="relative max-w-176 w-full flex flex-col-reverse gap-4"
  >
    <input
      aria-describedby="input-test-error"
      aria-invalid="false"
      class="peer z-1 text-md text-gray-600 font-normal border border-gray-300 caret-blue-base rounded-lg py-7 placeholder:text-gray-400 focus:outline-blue-base"
      id="input-test"
      style="padding-left: calc(1.6875rem); padding-right: 1rem;"
      type="text"
    />
    <span
      class="absolute top-18.5 left-8.5 text-md text-gray-400 font-normal z-1"
      data-testid="input-test-fixed-placeholder"
    >
      Placeholder
    </span>
    <label
      class="text-xs text-gray-500 uppercase focus:font-bold peer-focus:font-bold peer-focus:text-blue-base"
      for="input-test"
    >
      Title
    </label>
  </div>
</div>
`;
